f5lbaasdriver/test/tempest/tests/api/base.py

```diff
@@ -0,0 +1,434 @@
+# Copyright 2015, 2016 Rackspace Inc.
+# All Rights Reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+
+import time
+
+from oslo_log import log as logging
+from tempest.api.network import base
+from tempest import config
+from tempest.lib import exceptions
+
+from f5lbaasdriver.test.tempest.services.clients import \
+    l7policy_client
+from f5lbaasdriver.test.tempest.services.clients import \
+    plugin_rpc_client
+from neutron_lbaas.tests.tempest.v2.clients import \
+    health_monitors_client
+from neutron_lbaas.tests.tempest.v2.clients import listeners_client
+from neutron_lbaas.tests.tempest.v2.clients import \
+    load_balancers_client
+from neutron_lbaas.tests.tempest.v2.clients import members_client
+from neutron_lbaas.tests.tempest.v2.clients import pools_client
+
+CONF = config.CONF
+
+LOG = logging.getLogger(__name__)
+
+
+def _setup_client_args(auth_provider):
+    """Set up ServiceClient arguments using config settings."""
+    service = CONF.network.catalog_type or 'network'
+    region = CONF.network.region or 'regionOne'
```

Comment:
I guess this would only apply if the tempest conf network region is not provided.